src/Weeder/Main.hs

```diff
@@ -140,8 +219,34 @@ main = do
 --
 -- This will recursively find all files with the given extension in the given directories, perform
 -- analysis, and report all unused definitions according to the 'Config'.
-mainWithConfig :: String -> [FilePath] -> Bool -> Config -> IO (ExitCode, Analysis)
-mainWithConfig hieExt hieDirectories requireHsFiles weederConfig@Config{ rootPatterns, typeClassRoots, rootInstances, rootClasses } = do
+--
+-- Exits with one of the listed Weeder exit codes on failure.
+mainWithConfig :: String -> [FilePath] -> Bool -> Config -> IO ()
+mainWithConfig hieExt hieDirectories requireHsFiles = 
+  handleExits . mainWithConfig' hieExt hieDirectories requireHsFiles
+
+
+-- | 'mainWithConfig' without 'WeederException' handling
+mainWithConfig' :: String -> [FilePath] -> Bool -> Config -> IO ()
+mainWithConfig' hieExt hieDirectories requireHsFiles weederConfig = do
```

Comment:
I would use a more explicit name here. Otherwise we're making people have to hunt to the definition/documentation to work out what the `'` is doing.

```suggestion
mainWithConfigWithExceptions :: String -> [FilePath] -> Bool -> Config -> IO ()
mainWithConfigWithExceptions hieExt hieDirectories requireHsFiles weederConfig = do
```